zebra-chain/src/orchard/tree.rs

```diff
@@ -128,6 +126,13 @@ impl Hash for Root {
     }
 }
 
+impl PartialEq for Root {
+    fn eq(&self, other: &Self) -> bool {
+        // TODO: should we compare canonical forms here using `.to_repr()`?
```

Comment:
Is there a benefit to comparing the canonical forms here?

Reply:
If there is a non-canonical tree, then it could compare equal when hashed with `.to_repr()`, but unequal with `PartialEq`.

I think this is ok for now, because we always canonicalise our trees in Zebra. But I'm not sure enough, so I wanted to leave the question in a comment for later.